code/game/objects/structures/mannequin.dm

```diff
@@ -0,0 +1,1298 @@
+		SLOT_MANNEQUIN_ID,
+		)
+	obj_overlays = list(
+		MANNEQUIN_UNIFORM_LAYER,
+		MANNEQUIN_SHOES_LAYER,
+		MANNEQUIN_GLOVES_LAYER,
+		MANNEQUIN_EARS_LAYER,
+		MANNEQUIN_SUIT_LAYER,
+		MANNEQUIN_GLASSES_LAYER,
+		MANNEQUIN_BELT_LAYER,
+		MANNEQUIN_FACEMASK_LAYER,
+		MANNEQUIN_HEAD_LAYER,
+		MANNEQUIN_BACK_LAYER,
+		MANNEQUIN_ID_LAYER,
+		)
+
+	for(var/i = 1; i <= held_items.len; i++)
+		obj_overlays |= "[MANNEQUIN_HAND_LAYER]-[i]"
+
+	obj_overlays[MANNEQUIN_UNIFORM_LAYER]	= getFromPool(/obj/Overlays/uniform_layer)
+	obj_overlays[MANNEQUIN_ID_LAYER]		= getFromPool(/obj/Overlays/id_layer)
+	obj_overlays[MANNEQUIN_SHOES_LAYER]		= getFromPool(/obj/Overlays/shoes_layer)
+	obj_overlays[MANNEQUIN_GLOVES_LAYER]	= getFromPool(/obj/Overlays/gloves_layer)
+	obj_overlays[MANNEQUIN_EARS_LAYER]		= getFromPool(/obj/Overlays/ears_layer)
+	obj_overlays[MANNEQUIN_SUIT_LAYER]		= getFromPool(/obj/Overlays/suit_layer)
+	obj_overlays[MANNEQUIN_GLASSES_LAYER]	= getFromPool(/obj/Overlays/glasses_layer)
+	obj_overlays[MANNEQUIN_BELT_LAYER]		= getFromPool(/obj/Overlays/belt_layer)
+	obj_overlays[MANNEQUIN_BACK_LAYER]		= getFromPool(/obj/Overlays/back_layer)
+	obj_overlays[MANNEQUIN_FACEMASK_LAYER]	= getFromPool(/obj/Overlays/facemask_layer)
+	obj_overlays[MANNEQUIN_HEAD_LAYER]		= getFromPool(/obj/Overlays/head_layer)
+
+
+/obj/structure/mannequin/Destroy()
+	for(var/cloth in clothing)
+		if(clothing[cloth])
+			var/obj/item/cloth_to_drop = clothing[cloth]
+			cloth_to_drop.forceMove(loc)
+			clothing[cloth] = null
+	for(var/item in held_items)
+		if(held_items[item])
+			var/obj/item/item_to_drop = held_items[item]
+			item_to_drop.forceMove(loc)
+			held_items[item] = null
+	..()
+
+
+/obj/structure/mannequin/MouseDrop(var/mob/M)
+	..()
+	if(M != usr)
+		return
+	if(!Adjacent(M))
+		return
+	if(istype(M,/mob/living/silicon/ai))
+		return
+	show_inv(M)
+
+
+/obj/structure/mannequin/attack_hand(var/mob/user)
+	if(user.a_intent == I_HURT)
+		user.delayNextAttack(8)
+		user.visible_message("<span class='danger'>[user.name] kicks \the [src]!</span>", "<span class='danger'>You kick \the [src]!</span>")
+		health -= 2
+		healthcheck()
+	else
+		show_inv(user)
+
+
+/obj/structure/displaycase/attack_paw(mob/user as mob)
+	return attack_hand(user)
+
+
+/obj/structure/mannequin/attackby(var/obj/item/weapon/W,var/mob/user)
+	if(iswrench(W))
+		if(anchored)
+			to_chat(user, "<span class='info'>You start unsecuring \the [src] from \the [loc].</span>")
+		else
+			if(!istype(loc, /turf/simulated/floor))
+				return
+			to_chat(user, "<span class='info'>You start securing \the [src] to \the [loc].</span>")
+
+		spawn()
+			playsound(get_turf(src), 'sound/items/Ratchet.ogg', 100, 1)
+			if(do_after(user, src, 50))
+				anchored = !anchored
+				to_chat(user, "<span class='info'>[anchored ? "You successfully secure \the [src] to \the [loc]." : "You successfully unsecure \the [src] from \the [loc]."]")
+		return 1
+
+	attack_hand(user)
+
+
+/obj/structure/mannequin/examine(mob/user)
```

Comment:
Feel free to rewrite the entirety of inventorycode to be more modular if this is an issue, celt.